src/crimson/common/exception.h

```diff
@@ -18,8 +18,21 @@ class system_shutdown_exception final : public std::exception{
   }
 };
 
+class actingset_changed final : public std::exception{
+public:
+  actingset_changed(bool sp) : still_primary(sp) {}
+  const char* what() const noexcept final {
+    return "acting set changed";
+  }
+  bool is_primary() const {
+    return still_primary;
+  }
+private:
+  const bool still_primary = false;
+};
+
 template<typename Func, typename... Args>
-inline seastar::future<> handle_system_shutdown(Func&& func, Args&&... args)
+inline auto handle_system_shutdown(Func&& func, Args&&... args)
```

Comment:
nit, i still think it'd be nice if the return type could be explicitly noted. it helps the reader to understand the this function better.